bot/cogs/doc.py

```diff
@@ -0,0 +1,546 @@
+                description="This appears to be a generic page not tied to a specific symbol."
+            )
+
+        return discord.Embed(
+            title=f'`{symbol}`',
+            url=permalink,
+            description=f"```py\n{signature}```{description}"
+        )
+
+    async def get_all_packages(self) -> List[Dict[str, str]]:
+        """
+        Performs HTTP GET to get all packages from the website.
+
+        :return:
+        A list of packages, in the following format:
+        [
+            {
+                "package": "example-package",
+                "base_url": "https://example.readthedocs.io",
+                "inventory_url": "https://example.readthedocs.io/objects.inv"
+            },
+            ...
+        ]
+        `package` specifies the package name, for example 'aiohttp'.
+        `base_url` specifies the documentation root URL, used to build absolute links.
+        `inventory_url` specifies the location of the Intersphinx inventory.
+        """
+
+        async with self.bot.http_session.get(self.url, headers=self.headers) as resp:
+            return await resp.json()
+
+    async def get_package(self, package_name: str) -> Optional[Dict[str, str]]:
+        """
+        Performs HTTP GET to get the specified package from the documentation database.
+
+        :param package_name: The package name for which information should be returned.
+        :return:
+        Either a dictionary with information in the following format:
+        {
+            "package": "example-package",
+            "base_url": "https://example.readthedocs.io",
+            "inventory_url": "https://example.readthedocs.io/objects.inv"
+        }
+        or `None` if the site didn't returned no results for the given name.
+        """
+
+        params = {"package": package_name}
+
+        async with self.bot.http_session.get(
+            self.url, headers=self.headers, params=params
+        ) as resp:
+            package_data = await resp.json()
+            if not package_data:
+                return None
+            return package_data[0]
+
+    async def set_package(self, name: str, base_url: str, inventory_url: str) -> Dict[str, bool]:
+        """
+        Performs HTTP POST to add a new package to the website's documentation database.
+
+        :param name: The name of the package, for example `aiohttp`.
+        :param base_url: The documentation root URL, used to build absolute links.
+        :param inventory_url: The absolute URl to the intersphinx inventory of the package.
+
+        :return: The JSON response of the server, which is always:
+        {
+            "success": True
+        }
+        """
+
+        package_json = {
+            'package': name,
+            'base_url': base_url,
+            'inventory_url': inventory_url
+        }
+
+        async with self.bot.http_session.post(
+            self.url, headers=self.headers, json=package_json
+        ) as resp:
```

Comment:
These indents may seem unreadable to some users, try changing indentation.

Reply:
Seems pretty readable to me. What would you suggest instead?

Reply:
```python
async with self.bot.http_session.post(self.url, headers=self.headers, json=package_json) as resp:
    ...
# or 

async with self.bot.http_session.post(
        self.url, headers=self.headers, json=package_json) as resp:  # 8 indents to differ from code indent
    ...
# or 

async with self.bot.http_session.post(self.url, 
                                      headers=self.headers, 
                                      json=package_json) as resp:
    ...
```

basically, while your code works, it may show to inexperienced users as if the `)` might close the `async with` as it's on the same indent level as the `async with`.